src/main/java/org/assertj/core/internal/Bytes.java

```diff
@@ -63,4 +65,29 @@ protected boolean isGreaterThan(Byte value, Byte other) {
     return value > other;
   }
 
+  /**
+   * Asserts that the actual Byte value is even.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the actual value.
+   * @throws AssertionError if the actual value is {@code null}.
+   * @throws AssertionError if the actual value is negative.
+   */
+  public void assertIsEven(AssertionInfo info, Byte actual) {
+    Byte lastDigit = (byte) (actual & one());
+    assertIsZero(info, lastDigit);
```

Comment:
Although the `assertIsZero` check is correct, it will give a confusing error message:
```
"%nExpecting:%n <1>%nto be equal to:%n <0>%nbut was not."
``` 
The better approach is to define a specific error message for even assertion and one for odd.
I suggest:
```
"%nExpecting <%s> to be odd"
``` 

These error messages must be defined in `ShouldBeOdd`/`ShouldBeEven`, use `ShouldBeBase64`/`ShouldBeBase64_create_Test` for an implementation guide.

Having a specific error message requires to define `assertIsOdd`/`assertIsEven`,  where to do that?
* in `org.assertj.core.internal.Numbers` 
* in a new interface `WholeNumbers` with default methods

I have given a quick try to this second option, 
```
public interface WholeNumbers<NUMBER extends Number> {

  boolean isEven(NUMBER number);

  default boolean isOdd(NUMBER number) {
    return !isEven(number);
  }

  default void assertIsOdd(NUMBER number, AssertionInfo info) {
    if (!isOdd(number)) throw Failures.instance().failure(info, (number));
  }

  default void assertIsEven(NUMBER number, AssertionInfo info) {
    if (!isEven(number)) throw Failures.instance().failure(info, shouldBeEven(number));
  }
}
```
This approach needs to be evaluated, let me know if you need clarifications for that.